### DISCUSSION TITLE
package java.net.http does not exist

### DISCUSSION THREAD

**Discussion by wolled** [beep-projects/SystaPi#7] [Q&A] [answered]
I tried to execute the build.sh file on a rasbeperry and it claims that the java.net.http module is missing on the raspberry PI.
The installation of zulu java 11 went fine.....
Error: ./src/de/freaklamarsch/systarest/SystaRESTServer.java:17: error: package java.net.http does not exist
import java.net.http.HttpClient;
Any hint is very much appreciated.

**Comment by wolled** [comment, marked as answer]
I missed that I had two java versions installed. After I changed from from javac 1.8 to zulu 11 it worked :-)

**Reply by beep-projects** [reply]
Just to be sure, you ran the secondrun.sh on a raspberry that already was used for another project?
With a fresh Raspberry Pi OS, I would be surprised to have two java versions on the unit. (that is why I don't have any checks for the call to update-alternatives)

**Reply by wolled** [reply]
correct I did run the secondrun.sh an a RPI, and I was pretty sure its a fresh installtion... now when you ask.... I might have used it for another project and this came with java... so I can not confirm 100% its really fresh... sorry for that.
I am not fully done with the nic config.. but working on it...I am really courious to see systa sending the data captured by you nice project :-)

**Reply by beep-projects** [reply]
the NIC confi with setting up the fake paradigma.remote.portal DNS should also be done by secondrun.sh. The easiest way to set things up is to take a micro SD that you don't need for anything else and follow the install steps. That should be done in <30 minutes.
Nevertheless the trickiest part of it is to get the SystaComfort to send unencrypted messages. For that you need the Systa Service software.

**Reply by wolled** [reply]
thanks for all your hints. finally my NIC's are configured correctly, and the SystaComfort is sending the data also unencrypted.... I used the Lan_config tool to achive this....
Only my SystaComfort is not reconizing the "fake" URL correctly even it works. from another PC in the same local isolate LAN.I might need to restart my systacomfort II tomorrow.. and give it another try :-)

**Reply by beep-projects** [reply]
in order to retrigger the SystaComfort to obtain a new IP address from a DHCP server and to accept a new DNS for the portal, you just simply have to pull and reconnect the network cable

**Reply by wolled** [reply]
thanks for the info. funny enough this is working fine for any PC aquirring a DHCP adress from the Raspi, but not for the Systa Comfort. The systa comfort can not connect..... but is seeing a network but not getting an IP adress....  still checking why no IP adress is beeing assigned even when reconnecting the nic plug....

**Reply by wolled** [reply]
my network is now stable :-) and my systa gets an IP from my raspi.

unfortunately the API is not answering to API calls....
I am just wondering if my systa comfort II is supported by your nice API.... looking at the syslog of my raspi the follwoing is logged:

Jan 6 16:23:21 raspberrypi java[1011]: [FakeSystaWeb] trying to open DatagramSocket for UDP communication
Jan 6 16:23:21 raspberrypi java[1011]: [ParadigmaRESTAPI] start: called
Jan 6 16:23:21 raspberrypi java[1011]: [ParadigmaRESTAPI] start: FakeSystaWeb is already running, ignoring request
Jan 6 16:23:21 raspberrypi java[1011]: [FakeSystaWeb] UDP communication with Paradigma SystaComfort II started

for me all looks good :-) but when checking the status of the API I get the follwing:
calling: "http://192.168.178.112:1337/SystaREST/servicestatus" I get the follwing result,

connected | false
running | true
lastDataReceivedAt | "never"
packetsReceived | 61
paradigmaListenerIP | "192.168.1.1"
paradigmaListenerPort | 22460
paradigmaIP | "192.168.1.22"
paradigmaPort | 8002
loggingData | false
logFileSize | 60
logFilePrefix | "DataLogger"
logFileDelimiter | ";"
logFileRootPath | "/home/pi/logs"
logFilesWritten | 0
logBufferedEntries | 60

Any Idea what I can check in order to get some output when calling /SystaREST/servicestatus ? at the moment this page stay empty.

thansk a lot in advance.

**Reply by beep-projects** [reply]
Hmm, the status shows that it is receiving packets, but cannot use them. Are you sure that the traffic is not encrypted? If it is not encrypted, I would need a log from http://systapi:1337/SystaREST/enablelogging to check what the SystaComfort is sending. Or pcaps of the whole traffic.

Can you call http://systapi:1337/SystaREST/findsystacomfort ? That should give some info about the software versions that you are running.

**Reply by wolled** [reply]
its really strange. I have disabled the encryption with the LanConfig tool and have verified the setting on the remoteportal page. Here is shows that the traffic is not encrypted anymore thus I am confident that the traffic is "ok, or not encrypted anymore.... 

Calling http://systapi:1337/SystaREST/enablelogging shows an empty page in FF and an error in chrome ("this page dows not work properly....") 

Calling http://systapi:1337/SystaREST/findsystacomfort shows the following: 

SystaWebIP | "192.168.1.1"
SystaWebPort | 22460
DeviceTouchBcastIP | "192.168.1.255"
DeviceTouchBcastPort | 8001
deviceTouchInfoString | "SC2 1 192.168.1.22 255.255.255.0 192.168.1.1 SystaComfort-II\u00000 0809780002 0 V0.28 V1.00 2CBE9700XXXX"
unitIP | "192.168.1.22"
unitName | "SystaComfort-II"
unitId | "0809780002"
unitApp | 8
unitPlatform | 9
unitVersion | "1.2.2"
unitMajor | 120
unitMinor | 2
unitBaseVersion | "V0.28"
unitMac | "removed Mac address"
STouchAppSupported | false
DeviceTouchPort | -1
DeviceTouchPassword | "null"

I have two pcap files generated, one with the traffic when the systa is connected to remoteportal, and another one when systapi is connected.
I would be more then happy sharing the files. What would be the best way to exchange the files if you would like to have a look at it what would be just brilliant ?  unfortunatly a PN is not possible in git :-(

**Reply by beep-projects** [reply]
great, just send me an email to "the.beep.projects at gmail.com" or send me a message via facebook https://www.facebook.com/Beep-projects-105097291965997
